block0.go

```diff
@@ -40,26 +41,24 @@ func readBlock0(filename string) (block0, error) {
 	}
 	defer f.Close()
 
-	b0 := make([]byte, block0Size)
-
-	if _, err := f.Read(b0); err != nil {
-		return nil, err
-	}
-
-	return block0(b0), nil
+	var b0 [block0Size]byte
+	// The biggest offset we use is 1008, and read an uint64 from it,
+	// so we need at least 1008+8 bytes.
+	n, err := io.ReadAtLeast(f, b0[:], 1008+8)
```

Comment:
I disagree that io.ReadAtLeast is proper here.  Also, while the 1008 value is how the offsets add up, the 1048 value is the one that vim uses internally so I'm going to stick with that one (besides, this is 2018; we no longer need to quibble over 40 bytes)

Reply:
Ok, so you want io.ReadFull.